sample_annotator/clients/submission_portal_client.py

```diff
@@ -0,0 +1,185 @@
+        text_params = ['source_mat_id','env_package', 'agrochem_addition', 'al_sat_meth', 'crop_rotation', 'cur_vegetation', 'cur_vegetation_meth', 'heavy_metals', 'heavy_metals_meth', 'horizon_meth', 'link_class_info', 'link_climate_info', 'local_class', 'misc_param', 'local_class_meth', 'previous_land_use', 'soil_type', 'soil_type_meth', 'ph_meth', 'tot_org_c_meth', 'salinity_meth', 'store_cond', 'geo_loc_name', 'sieving', 'biotic_regm', 'air_temp_regm', 'climate_environment', 'gaseous_environment', 'humidity_regm', 'light_regm', 'watering_regm']
+        quantity_params = ['slope_aspect', 'al_sat', 'annual_precpt', 'annual_temp', 'season_precpt', 'season_temp', 'slope_gradient', 'soil_text_measure', 'temp', 'microbial_biomass', 'carb_nitro_ratio', 'org_matter', 'org_nitro', 'tot_carb', 'tot_nitro_content', 'tot_org_carb', 'tot_phosp', 'phosphate', 'salinity', 'samp_store_temp', 'size_frac_low', 'depth', 'size_frac_up', 'samp_size', 'alt']
+        controlled_term_params = ['env_broad_scale', 'env_local_scale', 'env_medium', 'experimental_factor', 'growth_facil', 'samp_mat_process', 'chem_administration']
+        float_params = ['elev', 'ph']
+        time_params = ['collection_date']
+        geoloc_params = ['lat_lon']
+
+
+        sampledata = response['metadata_submission']['sampleData']
+        db_lis = []
+        for super_keys in sampledata.keys():
+            current_lis = sampledata[super_keys]
+            db = nmdc.Database()
+            id_counter = 0
+            nmdc_ids = self.mint_nmdc_ids(len(current_lis))
+
+            for samp_dic in current_lis:
+                param_dic = {}
+                for key in samp_dic.keys():
+
+                    if(samp_dic[key] == 'nan' or samp_dic[key] == 'null' or samp_dic[key] == None):
+                        continue
+
+                    if (key in string_params):
+                        if (key == 'id'):
+                            param_dic['alternative_identifiers'] = samp_dic[key] #this vlue might need to be different than 'alternative_identifiers'
+                            param_dic[key] = nmdc_ids[id_counter]
+                            id_counter = id_counter + 1
+                            
+                        else:
+                            if(";" in samp_dic[key]):
+                                temp = samp_dic[key].split(";")
```

Comment:
The `split()` operation returns a list. Did you mean to pick the first element and use that as the id after you've done the split?

Reply:
There were a few issues with fields that had semicolon delimiters so if they were present i was converting them to a list. With the updated api output this isnt a problem anymore so ive removed this section of code.